addons/auth_totp/models/res_users.py

```diff
@@ -79,6 +79,61 @@ def _totp_try_setting(self, secret, code):
         _logger.info("2FA enable: SUCCESS for %s %r", self, self.login)
         return True
 
+    def get_totp_invite_url(self, user):
+        return '/web#action=auth_totp.action_activate_two_factor_authentication'
+
+    def totp_invite(self):
+        mails_to_send = self.env['mail.mail']
+        invite_template = self.env.ref('auth_totp.mail_template_totp_invite', raise_if_not_found=False)
+        for user in self.sudo().filtered(lambda user: not user.totp_secret):
+            ctx = {
+                'author_name': self.env.user.partner_id.name,
+                'url': self.get_totp_invite_url(user),
+            }
+            RenderMixin = self.env['mail.render.mixin'].with_context(**ctx)
+            subject = RenderMixin._render_template(invite_template.subject, 'res.users', self.ids, post_process=True)[user.id]
```

Comment:
Why not using send_email on Template model ?

Reply:
Up ? As normally send_email does everything you want here, including notification layout inclusion.

Reply:
Ah yes, veel beter !